### ISSUE TITLE
Otterscan fails to load wallet history and block transactions with anvil local fork

### ISSUE THREAD

**Issue by Nikotos** [foundry-rs/foundry#7881]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [x] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (6da2ff4c 2024-05-07T10:40:04.621682000Z)

### What command(s) is the bug in?

anvil --fork-url --fork-block-number

### Operating System

macOS (Apple Silicon)

### Describe the bug

it seems to be couple of bugs, even after their patches fixing it. (issue-7516) (https://github.com/foundry-rs/foundry/issues/7516)

latest foundry commit isn’t fixing this issue (6da2ff4c)

Long story short:
1) Otterscan can’t fetch the wallets history
2) Otterscan can’t fetch transactions in local fork blocks 

it seems to be caused by the behavior of the:
a) ots_searchTransactionsBefore
b) ots_searchTransactionsAfter
c) ots_getBlockTransactions

checked with several other developers, they have the same problem

**Comment by zerosnacks** [comment]
Hi @Nikotos thanks for the report

Would you be able to provide a minimal reproduction with steps?

**Comment by petscheit** [comment]
I am having a similar problem, once I try to fetch the transactions of a block, anvil crashes. 

I run: `anvil -f <MY_FORK_RPC> -p 6900`

and then: `docker run --rm -p 5100:80 --name otterscan -e ERIGON_URL="http://localhost:6900" otterscan/otterscan:latest`

I will open an issue in the foundry repo and cross-link it here.

**Comment by petscheit** [comment]
I am having a similar problem. When I access the block txs list via otterscan, anvil crashes.

I run: `anvil -f <MY_FORK_RPC> -p 6900`

and then: `docker run --rm -p 5100:80 --name otterscan -e ERIGON_URL="http://localhost:6900" otterscan/otterscan:latest`

Anvil crashes with the following error:

```
eth_getTransactionReceipt
thread 'tokio-runtime-worker' panicked at crates/anvil/src/eth/otterscan/api.rs:409:73:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I am running `anvil 0.2.0`

**Comment by zerosnacks** [comment]
For future reference: https://github.com/foundry-rs/foundry/blob/dc23de3f5a880c0487bc11fd60b9152f9d7b12bb/crates/anvil/src/eth/otterscan/api.rs#L409

**Comment by grandizzy** [comment]
> I am having a similar problem. When I access the block txs list via otterscan, anvil crashes.
> 

I am not able to reproduce anvil crash, otterscan docker started:

![image](https://github.com/user-attachments/assets/7fb11946-cce1-4056-b9dc-ced7aa27c26e)

and anvil keeps up even under heavy load

![image](https://github.com/user-attachments/assets/572d60ea-3015-47b1-821e-fc25c559b900)

**Comment by jenpaff** [comment]
closing as this seems to be fixed, please re-open if the issue still occurs
